File: collection/list/iter/iter.py

numbers = [1, 2, 3]
d = {"a": 1, "b": 2, "c": 3}

print("iterate list:")
for e in [1, 2, 3]:
    print(e)

print("iterate list with index:")
for i in range(len(numbers)):
    print(numbers[i], i)

print("iterate with both index and value")
for i, e in enumerate(numbers):
    print(e, i)

print("block is not pythonic, use regular function as argument instead:")


def larger_than_one(e):
    return e > 1


print(list(filter(larger_than_one, numbers)))

sum = 0
while sum < 3:
    sum += 1
print(sum)

File: collection/list/iter/iter.go
package main

import "fmt"

func main() {
	fmt.Println("iterate list")
	numbers := []int{10, 2, 3}

	for _, e := range numbers {
		fmt.Println(e)
	}

	fmt.Println("iterate with index and value")
	for i, e := range numbers {
		fmt.Println(e, i)
	}

	fmt.Println("iterate list with index")

	for i := 0; i < len(numbers); i++ {
		fmt.Println(numbers[i])
	}

	sum := 0
	fmt.Println("while loop")
	for sum < 3 {
		sum++
	}
	fmt.Println(sum)
}
